arcgis-ios-sdk-samples/Edit data/Edit feature attachments/AttachmentsTableViewController.swift

```diff
@@ -0,0 +1,174 @@
+            
+            if let error = error {
+                self.presentAlert(error: error)
+            }
+            
+            self.attachments = attachments ?? []
+            self.tableView.reloadData()
+        }
+    }
+    
+    private func deleteAttachment(_ attachment: AGSAttachment) {
+        feature?.delete(attachment) { [weak self] (error: Error?) in
+            if let error = error {
+                print(error)
+            } else {
+                self?.loadAttachments()
+            }
+        }
+    }
+    
+    // MARK: - Table view data source
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return attachments.count
+    }
+    
+    // MARK: - Table view delegate
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "AttachmentCell", for: indexPath)
+        
+        let attachment = attachments[indexPath.row]
+        cell.textLabel?.text = attachment.name
+        
+        cell.imageView?.image = UIImage(named: "CloudDownload", in: AGSBundle(), compatibleWith: nil)
+        cell.imageView?.contentMode = .scaleAspectFit
+        cell.imageView?.autoresizingMask = []
+        cell.imageView?.clipsToBounds = true
+        if attachment.hasFetchedData {
+            downloadAndSetAttachmentImageForCell(cell, at: indexPath)
+        }
+        
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath)!
+        downloadAndSetAttachmentImageForCell(cell, at: indexPath)
+        cell.setSelected(false, animated: true)
+    }
+    
+    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            let attachment = attachments[indexPath.row]
+            deleteAttachment(attachment)
+        }
+    }
+    
+    private func downloadAndSetAttachmentImageForCell(_ cell: UITableViewCell, at indexPath: IndexPath) {
+        let attachment = attachments[indexPath.row]
+        
+        if !attachment.hasFetchedData {
+            SVProgressHUD.show(withStatus: "Downloading attachment")
+        }
+        
+        attachment.fetchData { (data: Data?, error: Error?) in
+            
+            SVProgressHUD.dismiss()
+            
+            if let error = error {
+                print(error)
+            } else if let data = data,
+                let image = UIImage(data: data) {
+                cell.imageView?.image = image
+            }
+        }
+    }
+    
+    // MARK: - Actions
+    
+    @IBAction func doneAction() {
+        
+        guard let table = feature?.featureTable as? AGSServiceFeatureTable else {
+            dismiss(animated: true, completion: nil)
```

Comment:
I would strongly suggest not doing any work inside the `else` clause of a `guard` statement, other than perhaps logging. The whole purpose of a `guard` statement is to transfer control out of the current scope--bail early, bail often. Doing additional work before balling changes that and makes the whole function harder to reason about.

I would suggestion using an `if` statement:

```swift
if let table = feature?.featureTable as? AGSServiceFeatureTable {
    // Apply edits to the table…
} else {
    dismiss(animated: true)
}
```

Or you could break it up into two different functions:

```swift
@IBAction func doneAction() {
    //show progress hud
    SVProgressHUD.show(withStatus: "Applying edits")
    
    applyEdits { [weak self] (error) in
        
        //dismiss progress hud
        SVProgressHUD.dismiss()
        
        guard let self = self else {
            return
        }
        
        if let error = error {
            self.presentAlert(error: error)
        }
        self.dismiss(animated: true)
    }
    
}

func applyEdits(completion: @escaping (_ error: Error?) -> Void) {
    if let table = feature?.featureTable as? AGSServiceFeatureTable {
        table.applyEdits { completion($1) }
    } else {
        completion(nil)
    }
}
```